sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateProgressDao.java

```diff
@@ -52,12 +56,14 @@ public Progress getManaged(long id) {
 
   @Override
   public List<Progress> listByUserId(long id) {
-    Criteria criteria = currentSession().createCriteria(ProgressImpl.class);
-    criteria.createAlias("user", "u")
-        .add(Restrictions.eq("u.userId", id));
-    criteria.setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY);
+    QueryBuilder<?, ProgressImpl> builder =
+        new QueryBuilder<>(currentSession(), ProgressImpl.class, Object[].class, Criteria.DISTINCT_ROOT_ENTITY);
+    Join<ProgressImpl, UserImpl> userJoin = builder.getJoin(builder.getRoot(), ProgressImpl_.user);
+    builder.addPredicate(builder.getCriteriaBuilder().equal(userJoin.get(UserImpl_.userId), id));
+    builder.setColumns(builder.getRoot());
+
     @SuppressWarnings("unchecked")
-    List<Progress> results = criteria.list();
+    List<Progress> results = (List<Progress>) builder.getResultList();
```

Comment:
I think you can use `Progress.class` instead of `Object[].class` and then remove the `setColumns` and the cast?